doc/modules/clustering.rst

```diff
@@ -1582,3 +1582,59 @@ Drawbacks
  *  Caliński, T., & Harabasz, J. (1974). "A dendrite method for cluster
     analysis". Communications in Statistics-theory and Methods 3: 1-27.
     `doi:10.1080/03610926.2011.560741 <http://dx.doi.org/10.1080/03610926.2011.560741>`_.
+
+.. _contingency_matrix:
+
+Contingency Matrix
+----------------------
+
+Contingency matrix (:func:`sklearn.metrics.cluster.contingency_matrix`)
+reports the intersection cardinality for every true/predicted cluster pair.
+The contingency matrix provides sufficient statistics for all clustering
+metrics where the samples are independent and identically distributed and
+one doesn't need to account for some instances not being clustered.
+
+Here is an example::
+
+   >>> from sklearn.metrics.cluster import contingency_matrix
+   >>> x = ["true cluster 1", "true cluster 1", "true cluster 1",
+   ...      "true cluster 2", "true cluster 2", "true cluster 2"]
+   >>> y = ["pred cluster 1", "pred cluster 1", "pred cluster 2",
+   ...      "pred cluster 2", "pred cluster 3", "pred cluster 3"]
+   >>> contingency_matrix(x, y)
+   array([[2, 1, 0],
+          [0, 1, 2]])
+
+The first row of output array indicates that there are three samples labelled
+'true cluster 1', two of which are also labelled 'pred cluster 1', one is labelled
+'pred cluster 2' while none is labelled 'pred cluster 3'. Similarly the second row
+indicates there are three samples labelled 'true cluster 2' of which none is labelled
+'pred cluster 1', one is labelled 'pred cluster 2' and two are labelled
+'pred cluster 3'.
+
+Confusion matrix is square matrix and the order of columns and rows is significant.
+
+
+Advantages
+~~~~~~~~~~
+
+- The table allows us to compare at glance the proportion of samples with
+  labels.
+
+- Can help us detect if random variables are independent or not. If the
+  proportionsof individuals in the different columns vary significantly
+  between rows (or vice versa), we say that there is a contingency between the
+  two variables. In other words, the two variables are not independent. If
+  there is no contingency, we say that the two variables are independent.
+
+Drawbacks
+~~~~~~~~~
+
+- Higher order contingency tables are difficult to represent visually.
```

Comment:
What do you mean by higher order? We're only dealing with 2d here

Reply:
In general I think contingency table can have more than 2d, right? Or you stating that contingency table always is 2d (not just in scikit-learn)? 
Though here I intended to state a general limitation of contingency tables. I took this from wikpedia article https://en.wikipedia.org/wiki/Contingency_table#Example (last paragraph, second line).

Reply:
We are talking about it in the context of clustering evaluation where, yes, I can *imagine* it being used with more than one set of predictions, but it is certainly not supported in scikit-learn context. To talk of higher dimensions in a user guide will rarely benefit and will easily confuse.